### PR TITLE
[WasmFS] Support file embedding

### PR DESCRIPTION
**Pull request by kripken**
This practically worked since #16050, but the JS side needs to support a
missing parent, which is how the file packager emits it.

Fixes https://github.com/emscripten-core/emscripten/issues/16014

### CODE REVIEW

**Review by tlively — APPROVED** [review]

**Comment by kripken** [comment]
Minor issue, we had code paths that support both passing `name,null` and `null,name` to give just a name and not a pair of `parent,name`. Rather than support two ways of doing it, I standardized on one of them. This should fix the broken wasmfs test in the browser suite here.

**Review by tlively — COMMENTED** [review]
It seems stranger to provide `parent` but not `name` than it would be to provide `name` but not `parent`. Would it be feasible to standardize in the other direction?

**Comment by kripken** [comment]
I think of it as `name, null` being "one param is passed, not two, so interpret the only one as the full path". We have similar things elsewhere in JS, where the dynamic nature makes it possible to consider arguments as "optional" that way. But I don't feel strongly, if you still disagree?

**Comment by tlively** [comment]
If it's consistent and idiomatic, then lgtm, but it definitely feels weird! Sometimes the `parent` parameter is actually a parent, but other times it's actually the name while the `name` parameter goes unused.

**Comment by kripken** [comment]
Hmm, I looked around a little for an example in our codebase, and can't find one atm... maybe it's not as common as I thought. I'll flip it around as you suggest.

**Review by tlively — APPROVED** [review]

**Comment by kripken** [comment]
Oh, I missed the example, which was just a few lines above this, in `createPreloadedFile` (as opposed to `createDataFile` which is modified here). That has the logic the first way I did it, and in fact tests break if it does not match between the two places. So I'll go back to the previous way...

File: tools/file_packager.py [hunk 1]
```diff
@@ -649,7 +649,7 @@ def generate_js(data_target, data_files, metadata):
         var content = HEAPU32[start32++];
         var name = UTF8ToString(name_addr)
         // canOwn this data in the filesystem, it is a slice of wasm memory that will never change
-        Module['FS_createDataFile'](undefined, name, HEAP8.subarray(content, content + len), true, true, true);
+        Module['FS_createDataFile'](name, null, HEAP8.subarray(content, content + len), true, true, true);
```
Comment (sbc100) [inline tools/file_packager.py:652]: This seems odd to me too.  The way I handled this in #16050 I guess made it work both wasy:

```
-      var path = name ? PATH.join2(typeof parent === 'string' ? parent : FS.getPath(parent), name) : parent;
+      var path = name;
+      if (parent) {
+        parent = typeof parent === 'string' ? parent : FS.getPath(parent);
+        path = name ? PATH.join2(parent, name) : parent;
+      }
```

**Pull request merged**